Sources/FoundationEssentials/Data/Data+Base64.swift

```diff
@@ -0,0 +1,374 @@
+//
+//===----------------------------------------------------------------------===//
+
+private enum Base64Error: Error {
+    case invalidElementCount
+    case cannotDecode
+}
+
+@available(macOS 10.10, iOS 8.0, watchOS 2.0, tvOS 9.0, *)
+extension Data {
+
+    // MARK: - Init from base64
+
+    /// Initialize a `Data` from a Base-64 encoded String using the given options.
+    ///
+    /// Returns nil when the input is not recognized as valid Base-64.
+    /// - parameter base64String: The string to parse.
+    /// - parameter options: Encoding options. Default value is `[]`.
+    public init?(
+        base64Encoded base64String: consuming String,
+        options: Base64DecodingOptions = []
+    ) {
+        do {
+            self = try base64String.withUTF8 {
+                // String won't pass an empty buffer with a `nil` `baseAddress`.
+                try Data(
+                    decodingBase64: BufferView(unsafeBufferPointer: $0)!,
+                    options: options
+                )
+            }
+        }
+        catch {
+            precondition(error is Base64Error)
+            return nil
+        }
+    }
+
+
+    /// Initialize a `Data` from a Base-64, UTF-8 encoded `Data`.
+    ///
+    /// Returns nil when the input is not recognized as valid Base-64.
+    ///
+    /// - parameter base64Data: Base-64, UTF-8 encoded input data.
+    /// - parameter options: Decoding options. Default value is `[]`.
+    public init?(
+        base64Encoded base64Data: borrowing Data,
+        options: Base64DecodingOptions = []
+    ) {
+        do {
+            self = try base64Data.withBufferView {
+                try Data(decodingBase64: $0, options: options)
+            }
+        }
+        catch {
+            precondition(error is Base64Error)
+            return nil
+        }
+    }
+
+    private init(decodingBase64 bytes: borrowing BufferView<UInt8>, options: Base64DecodingOptions = []) throws {
+        if !bytes.count.isMultiple(of: 4) &&
+           !options.contains(.ignoreUnknownCharacters) {
+            throw Base64Error.invalidElementCount
+        }
+
+        // Every 4 valid ASCII bytes maps to 3 output bytes: (bytes.count * 3)/4
+        let capacity = (bytes.count * 3) >> 2
+        // A non-trapping version of the calculation goes like this:
+        // let (q, r) = bytes.count.quotientAndRemainder(dividingBy: 4)
+        // let capacity = (q * 3) + (r==0 ? 0 : r-1)
+        self = try Data(
+            capacity: capacity,
+            initializingWith: { //FIXME: should work with borrowed `bytes`
+                [bytes = copy bytes] in
+                try Data.base64DecodeBytes(bytes, &$0, options: options)
+            }
+        )
+    }
+
+    // MARK: - Create base64
+
+    /// Returns a Base-64 encoded string.
+    ///
+    /// - parameter options: The options to use for the encoding. Default value is `[]`.
+    /// - returns: The Base-64 encoded string.
+//    @available(macOS 11, *)
```

Comment:
What's the plan with this availability annotation?

The existing one is again `@inlinable`

Reply:
This one turned out to be unnecessary, and I forgot to delete it. The underlying `String` initializer requires macOS 11.